single_include/catch.hpp

```diff
@@ -8000,19 +8000,17 @@ namespace Catch {
         getResultCapture().pushScopedMessage( m_info );
     }
 
-#if defined(_MSC_VER)
-#pragma warning(push)
-#pragma warning(disable:4996) // std::uncaught_exception is deprecated in C++17
+// https://github.com/catchorg/Catch2/issues/1162
+#if __cplusplus == 201703L
+using uncaught_exceptions = std::uncaught_exceptions
```

Comment:
1. redefining language identifiers is a bad idea (and this might lead to weird clashes). I would use `catch_uncaught_exceptions`
2. Try and centralise this in a header shared by all instead of defining it three times in a row.